src/datepicker/datepicker.js

```diff
@@ -162,14 +170,82 @@ angular.module('ui.bootstrap.datepicker', ['ui.bootstrap.dateparser', 'ui.bootst
       ngModelCtrl.$render();
     } else {
       self.activeDate = date;
+      self.calculateDateinterval();
       $scope.datepickerMode = self.modes[self.modes.indexOf($scope.datepickerMode) - 1];
     }
   };
 
+  this.clearIntervalVars = function () {
+    self.ignoreDateCompare = false;
+    self.otherMonthSelectedDate = undefined;
+  };
+
+  this.calculateDateinterval = function () {
+    var year = self.activeDate.getFullYear();
+    var month = self.activeDate.getMonth();
+    if (typeof ngModelCtrl.$modelValue !== 'undefined' && ngModelCtrl.$modelValue !== null) {
+      var prevValue;
+      if (typeof ngModelCtrl.$modelValue === 'string' || ngModelCtrl.$modelValue instanceof String) {
```

Comment:
Hi,

When I've resolved the issues, what format should I use for the commit message?
Should it be: refactor(datepicker): <msg>

Regards,
Olle Thalén


> 25 sep 2015 kl. 20:27 skrev Wesley Cho <notifications@github.com>:
> 
> In src/datepicker/datepicker.js:
> 
> >        $scope.datepickerMode = self.modes[self.modes.indexOf($scope.datepickerMode) - 1];
> >      }
> >    };
> >  
> > +  this.clearIntervalVars = function () {
> > +    self.ignoreDateCompare = false;
> > +    self.otherMonthSelectedDate = undefined;
> > +  };
> > +
> > +  this.calculateDateinterval = function () {
> > +    var year = self.activeDate.getFullYear();
> > +    var month = self.activeDate.getMonth();
> > +    if (typeof ngModelCtrl.$modelValue !== 'undefined' && ngModelCtrl.$modelValue !== null) {
> > +      var prevValue;
> > +      if (typeof ngModelCtrl.$modelValue === 'string' || ngModelCtrl.$modelValue instanceof String) {
> Use angular.isString instead
> 
> —
> Reply to this email directly or view it on GitHub.
>